pkg/operator/podsecurityreadinesscontroller/conditions.go

```diff
@@ -28,24 +30,30 @@ var (
 )
 
 type podSecurityOperatorConditions struct {
-	violatingOpenShiftNamespaces    []string
-	violatingRunLevelZeroNamespaces []string
-	violatingCustomerNamespaces     []string
+	violatingOpenShiftNamespaces      []string
+	violatingRunLevelZeroNamespaces   []string
+	violatingCustomerNamespaces       []string
+	violatingDisabledSyncerNamespaces []string
 }
 
-func (c *podSecurityOperatorConditions) addViolation(name string) {
-	if runLevelZeroNamespaces.Has(name) {
-		c.violatingRunLevelZeroNamespaces = append(c.violatingRunLevelZeroNamespaces, name)
+func (c *podSecurityOperatorConditions) addViolation(ns *corev1.Namespace) {
+	if runLevelZeroNamespaces.Has(ns.Name) {
+		c.violatingRunLevelZeroNamespaces = append(c.violatingRunLevelZeroNamespaces, ns.Name)
 		return
 	}
 
-	isOpenShift := strings.HasPrefix(name, "openshift")
+	isOpenShift := strings.HasPrefix(ns.Name, "openshift")
```

Comment:
you shoulnd't need to treat the prefix differently in `isNSControlled()` if you return here